airflow/executors/base_executor.py

```diff
@@ -436,11 +440,14 @@ def fail(self, key: TaskInstanceKey, info=None) -> None:
                 component="BaseExecutor",
                 parent_sc=gen_context(trace_id=trace_id, span_id=span_id),
             ) as span:
-                span.set_attribute("dag_id", key.dag_id)
-                span.set_attribute("run_id", key.run_id)
-                span.set_attribute("task_id", key.task_id)
-                span.set_attribute("try_number", key.try_number)
-                span.set_attribute("error", True)
+                span.set_attributes(
+                    {
+                        "dag_id": key.dag_id,
+                        "run_id": key.run_id,
+                        "try_number": key.try_number,
+                        "error": True,
```

Comment:
1. seems you lost task_id
2. should this be pushed down into `change_state`?  do we want to create a span every time we change state? i don't even know what a span is 🤦

Reply:
Good catch on (1), added it.

For (2) Unsure, keeping it out of scope of this pR